pkg/secrets/secrets.go

```diff
@@ -21,3 +49,44 @@ func ValidateCredentials(secret *v1.Secret) error {
 		return errors.Errorf("Unsupported type '%s' for secret '%s:%s'", string(secret.Type), secret.Namespace, secret.Name)
 	}
 }
+
+func getLocationType(secret *v1.Secret) (repositoryserver.RepositoryServerSecret, error) {
+	var locationType []byte
+	var ok bool
+	if secret == nil {
+		return nil, errors.New("Secret is Nil")
+	}
+
+	if locationType, ok = (secret.Data[LocationTypeKey]); !ok {
+		return nil, errors.Errorf("secret '%s:%s' does not have required field %s", secret.Namespace, secret.Name, LocationTypeKey)
+	}
+
+	switch string(locationType) {
+	case storage.LocTypeS3:
+		return repositoryserver.NewAWSLocation(secret), nil
+	case storage.LocTypeAzure:
+		return repositoryserver.NewAzureLocation(secret), nil
+	default:
+		return nil, errors.Errorf("Unsupported location type '%s' for secret '%s:%s'", locationType, secret.Namespace, secret.Name)
+	}
+}
+
+func ValidateRepositoryServerSecret(repositoryServerSecret *v1.Secret) error {
+	var secret repositoryserver.RepositoryServerSecret
+	var err error
+
+	switch string(repositoryServerSecret.Type) {
+	case LocationSecretType:
+		secret, err = getLocationType(repositoryServerSecret)
+		if err != nil {
+			return err
+		}
+	case RepositoryPasswordSecretType:
```

Comment:
should these names be changed to `SecretTypeRepositoryPassword` and `SecretTypeLocation` instead. I have seen the format that gets used in these scenarios is `objcetTypeValue`.
For example `PodScheduled`, `PodInitialized` etc.

Or the other options would be
```
type SecretType string

var Location SecretType = "location"

switch string(repositoryServerSecret.Type) {
	case Location:
```

Reply:
makes sense, I have defined constants as
```
Location v1.SecretType = "secrets.kanister.io/storage-location"
// RepositoryPasswordSecretType represents the kopia repository passowrd secret type
RepositoryPassword v1.SecretType = "secrets.kanister.io/kopia-repository/password"
```
and used them directly. Would this work?